File: Dddml.Wms.HttpServices.JavaClient/src/org/dddml/wms/domain/RoleStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class RoleStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "roleId",
            "version",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private RoleStateEventId value;

    public RoleStateEventIdFlattenedDto()
    {
        this(new RoleStateEventId());
    }

    public RoleStateEventIdFlattenedDto(RoleStateEventId value)
    {
        this.value = value;
    }

    public RoleStateEventId toRoleStateEventId()
    {
        return this.value;
    }


    public String getRoleId()
    {
        return this.value.getRoleId();
    }

    public void setRoleId(String roleId)
    {
        this.value.setRoleId(roleId);
    }

    public Long getVersion()
    {
        return this.value.getVersion();
    }

    public void setVersion(Long version)
    {
        this.value.setVersion(version);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        RoleStateEventIdFlattenedDto other = (RoleStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Dddml/Wms/Domain/RoleStateEventIdFlattenedDto.php
<?php

namespace Dddml\Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\FlattenedDtoInterface;

class RoleStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'roleId' => 'string',
            'version' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var RoleStateEventId
     */
    private $value;

    /**
     * @param RoleStateEventId $value
     */
    public function __construct(RoleStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new RoleStateEventId();
        }
    }

    /**
     * @return RoleStateEventId
     */
    public function toRoleStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getRoleId()
    {
        return $this->value->getRoleId();
    }

    /**
     * @param string $roleId
     */
    public function setRoleId($roleId)
    {
        $this->value->setRoleId($roleId);
    }

    /**
     * @return Long
     */
    public function getVersion()
    {
        return $this->value->getVersion();
    }

    /**
     * @param Long $version
     */
    public function setVersion($version)
    {
        $this->value->setVersion($version);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getRoleId(),
            $this->getVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return RoleStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setRoleId($pValues[0]);
        $this->setVersion($pValues[1]);
        return $this;
    }

}

